tests/pipelines/test_pipelines_automatic_speech_recognition.py

```diff
@@ -266,6 +262,49 @@ def test_torch_whisper(self):
         output = speech_recognizer([filename], chunk_length_s=5, batch_size=4)
         self.assertEqual(output, [{"text": " A man said to the universe, Sir, I exist."}])
 
+    @require_torch
+    @slow
+    def test_whisper_timestamp_prediction(self):
+        processor = WhisperProcessor.from_pretrained("openai/whisper-tiny")
+        model = WhisperForConditionalGeneration.from_pretrained("openai/whisper-tiny")
+        model.config.forced_decoder_ids = processor.get_decoder_prompt_ids(
+            task="transcribe", language="en", no_timestamps=False
+        )
+        model.config.begin_suppress_tokens += [50363]
+        ds = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation").sort("id")
+        array = np.concatenate(
+            [ds[40]["audio"]["array"], ds[41]["audio"]["array"], ds[42]["audio"]["array"], ds[43]["audio"]["array"]]
+        )
+        pipe = pipeline(
+            "automatic-speech-recognition",
+            model=model,
+            tokenizer=processor.tokenizer,
+            feature_extractor=processor.feature_extractor,
+            forced_bos_token_id=None,
+            max_new_tokens=448,
+        )
+        output = pipe(array, return_timestamps=True, chunk_length_s=30, stride_length_s=[15, 0], device=0)
+        # fmt: off
+        EXPECTED_OUTPUT = {
```

Comment:
Let's put that directly in the test, to remove the indirection. And I'm pretty sure black can make it more readable we don't need `fmt: off` here.